File: Logic Building Problems/Easy Problems/Check for Prime Number/Different Methods to Check for Prime Number/method2.js
// Function to check whether a number is prime or not
function isPrime(n) {

    // Numbers less than or equal to 1 are not prime
    if (n <= 1)
        return false;

    // Check divisibility from 2 to the square root of n
    for (let i = 2; i <= Math.sqrt(n); i++)
        if (n % i === 0) 
            return false;

    // If no divisors were found, n is prime
    return true;
}

// Driver Code
let n = 11; 
if(isPrime(n))
	console.log("true");
else
	console.log("false");

File: Logic Building Problems/Easy Problems/Check for Prime Number/Different Methods to Check for Prime Number/method2.cpp
#include <iostream>
#include <cmath>
using namespace std;

// Function to check whether a number is prime or not
bool isPrime(int n) {

    // Numbers less than or equal to 1 are not prime
    if (n <= 1)
        return false;

    // Check divisibility from 2 to the square root of n
    for (int i = 2; i <= sqrt(n); i++)
        if (n % i == 0)  
            return false;

    // If no divisors were found, n is prime
    return true;
}

int main() {
    int n = 11;  
  	if(isPrime) cout << "true";
  	else cout<<"false";
    return 0;
}